DatadogSessionReplay/Sources/Recorder/RecordingTrigger.swift

```diff
@@ -0,0 +1,62 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+#if os(iOS)
+import DatadogInternal
+import UIKit
+
+/// Orchestrates the process of triggering next snapshot recording.
+/// It has 2 recording triggers: `layoutSubviews` and touch event, which are done by using swizzling.
+internal protocol RecordingTriggering {
+    func startWatchingTriggers()
+    func stopWatchingTriggers()
+
+    var delegate: RecordingTriggerDelegate? { get set }
+}
+
+/// Delegate to notify about recording trigger.
+internal protocol RecordingTriggerDelegate: AnyObject {
+    func didTrigger()
+}
```

Comment:
/suggestion I'm not a fan of delegate property in a protocol, because you cannot enforce `weak` from the protocol property. So one could assume it's a retain while it is not in the concrete type. Instead, we could a simple callback such as
```swift
internal protocol RecordingTriggering {
    func startWatchingTriggers(_ callback: @escaping () -> Void)
    func stopWatchingTriggers()
}
```
Would that work?